### PR TITLE
Switch to the now-built-in Marshal APIs from MarshalEx

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Now that our APIs are built-in to the runtime version we build and run against, we can switch to using the Marshal APIs directly instead of using the MarshalEx type.

Question for reviewers: Do we want to keep the infra around MarshalEx still in the experiment for other interop generators we may want to experiment with, or do we want to remove it?

### CODE REVIEW

**Review by AaronRobinsonMSFT — APPROVED** [review]
Looks awesome!

**Comment by jkoritzinsky** [comment]
Looks like I forgot to update the reference assemblies that we build against. Lets see if I can figure out a clean way to do that.

File: DllImportGenerator/Ancillary.Interop/Ancillary.Interop.csproj [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <AssemblyName>Microsoft.Interop.Ancillary</AssemblyName>
-    <TargetFramework>net5.0</TargetFramework>
+    <TargetFramework>net6.0</TargetFramework>
```
Comment (elinor-fung) [inline DllImportGenerator/Ancillary.Interop/Ancillary.Interop.csproj:5]: I think you should be able to remove the `UseCustomRuntimeVersion=false` below now.

**Review by elinor-fung — APPROVED** [review]
I'm fine keeping `MarshalEx` for now - at least until we actually get through trying to convert NETCoreApp. Could you put a comment somewhere in that `MarshalEx` class to point out that it should be used with `TypeNames.MarshalEx` in any generated code to allow configuration?

**Pull request merged**